scripts/build.ps1

```diff
@@ -0,0 +1,75 @@
+#########################################################
+# Script to build different parts of OpenRCT2
+#########################################################
+[CmdletBinding()]
+param (
+    [Parameter(Mandatory = $false, Position = 1)]
+    [string]$Task = "openrct2",
+
+    [Parameter(Mandatory = $false)]
+    [string]$Configuration = "Release",
+
+    [Parameter(Mandatory = $false)]
+    [switch]$Rebuild = $false
+)
+
+# Get paths
+$scriptPath   = $Script:MyInvocation.MyCommand.Path
+$scriptsPath  = Split-Path $scriptPath
+$rootPath     = Split-Path $scriptsPath
+$binPath      = Join-Path $rootPath "bin"
+$openrct2Path = Join-Path $binPath "openrct2.exe"
+
+function Build-Data()
+{
+    Write-Host "Copying data to bin..." -ForegroundColor Cyan
+    New-Item -Force -ItemType Directory $binPath > $null
+    Copy-Item -Force -Recurse "$rootPath\data" $binPath
+    return 0
+}
+
+function Build-OpenRCT2()
+{
+    Write-Host "Building OpenRCT2 ($Configuration)..." -ForegroundColor Cyan
+    
+    $target = ""
+    if ($Rebuild)
+    {
+        $target = "/t:rebuild"
+    }
+    msbuild $rootPath\openrct2.sln /p:Configuration=$Configuration /p:Platform=Win32 $target /v:minimal | Write-Host
+    return $LASTEXITCODE
+}
+
+function Build-G2()
+{
+    # Check if OpenRCT2 binary exists
+    if (-not (Test-Path $openrct2Path))
+    {
+        Write-Host "You must build OpenRCT2 first before g2.dat can be built." -ForegroundColor Red
```

Comment:
With CMake (as it is common in Linux projects), when a target requires some other target, we simply set the dependency chain and let `make` worry about that dependency being resolved. In other words: on Linux I would expect `make g2` to figure out it depends on target `openrct2`, build it if not current, then proceed to making target `g2`.

Is this approach (to bail out) common on Windows?